File: Kattis/solutions/closestSums/closestSums.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
  int n, caseNum = 1;
  while (cin >> n) {
    int m;
    vector<int> input(n);
    for (int i = 0; i < n; i++) {
      cin >> input[i];
    }

    sort(input.begin(), input.end());

    cout << "Case " << caseNum++ << ":\n";

    cin >> m;
    for (int i = 0; i < m; i++) {
      int query, closestSum = -1e9;
      cin >> query;

      for (int j = 0; j < n - 1; j++) {
        for (int k = j + 1; k < n; k++) {
          if (abs(input[j] + input[k] - query) < abs(closestSum - query)) {
            closestSum = input[j] + input[k];
          }
        }
      }

      cout << "Closest sum to " << query << " is " << closestSum << ".\n";
    }
  }

  return 0;
}

File: Kattis/solutions/closestSums/closestSums.java
import java.util.*;

public class closestsums {
  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);
    int n, caseNum = 1;

    while (scan.hasNext()) {
      n = scan.nextInt();
      int m;
      ArrayList<Integer> input = new ArrayList<>();
      for (int i = 0; i < n; i++) {
        input.add(scan.nextInt());
      }

      Collections.sort(input);

      System.out.println("Case " + caseNum++ + ":");

      m = scan.nextInt();
      for (int i = 0; i < m; i++) {
        int query = scan.nextInt(), closestSum = -1000000000;

        for (int j = 0; j < n - 1; j++) {
          for (int k = j + 1; k < n; k++) {
            if (Math.abs(input.get(j) + input.get(k) - query) < Math.abs(closestSum - query)) {
              closestSum = input.get(j) + input.get(k);
            }
          }
        }

        System.out.println("Closest sum to " + query + " is " + closestSum + ".");
      }
    }

    scan.close();
  }
}